File: app/src/main/java/fr/mpau/tools/MinMaxFilter.java
package fr.mpau.tools;

import android.text.InputFilter;
import android.text.Spanned;

/**
 * Classe de filtre pour imposer un range possible d'un EditText de type Number
 * <p>
 * Author: Jonathan B.
 * Created: 17/01/2018
 */


public class MinMaxFilter implements InputFilter {

    /**
     * Attributs
     */
    private int mIntMin;
    private int mIntMax;

    /**
     * Constructeur
     *
     * @param minValue int
     * @param maxValue int
     */
    public MinMaxFilter(int minValue, int maxValue) {
        this.mIntMin = minValue;
        this.mIntMax = maxValue;
    }

    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        try {
            int input = Integer.parseInt(dest.toString() + source.toString());
            if (isInRange(mIntMin, mIntMax, input))
                return null;
        } catch (NumberFormatException nfe) {
        }
        return "";
    }

    /**
     * Détermine si la valeur est dans le range
     *
     * @param min   int
     * @param max   int
     * @param value int
     * @return boolean
     */
    private boolean isInRange(int min, int max, int value) {
        return max > min ? value >= min && value <= max : value >= max && value <= min;
    }
}

File: src/main/java/fr/mpau/tool/MinMaxFilter.kt
package fr.mpau.tool

import android.text.InputFilter
import android.text.Spanned

/**
 * Tool - Classe de filtre pour imposer un range possible d'un EditText de type Number
 *
 * Author: Jonathan
 * Created: 17/01/2018
 * Last Updated: 24/01/2025
 */
class MinMaxFilter(private val mIntMin: Int, private val mIntMax: Int) : InputFilter {

    /**
     * Filter
     *
     * @param source CharSequence
     * @param start Int
     * @param end Int
     * @param dest Spanned
     * @param dstart Int
     * @param dend Int
     * @return CharSequence?
     */
    override fun filter(
        source: CharSequence,
        start: Int,
        end: Int,
        dest: Spanned,
        dstart: Int,
        dend: Int
    ): CharSequence? {
        try {
            val input = (dest.toString() + source.toString()).toInt()
            if (isInRange(mIntMin, mIntMax, input)) return null
        } catch (_: NumberFormatException) {
        }
        return ""
    }

    /**
     * Détermine si la valeur est dans le range
     *
     * @param min Int
     * @param max Int
     * @param value Int
     * @return Boolean
     */
    private fun isInRange(min: Int, max: Int, value: Int): Boolean {
        return if (max > min) value in min..max else value in max..min
    }

}